plugins/inputs/ping/ping.go

```diff
@@ -171,14 +171,24 @@ func (p *Ping) args(url string) []string {
 	// Build the ping command args based on toml config
 	args := []string{"-c", strconv.Itoa(p.Count), "-n", "-s", "16"}
 	if p.PingInterval > 0 {
-		args = append(args, "-i", strconv.FormatFloat(p.PingInterval, 'f', 1, 64))
```

Comment:
I think we can just change this to use the -1 precision:
```
args = append(args, "-i", strconv.FormatFloat(p.PingInterval, 'f', -1, 64))
```

It still will require users to specify an even value, but I think it is better than switching on arch.